module-3/lab-supervised-learning-feature-extraction/your-code/main.ipynb

```diff
@@ -12,7 +12,7 @@
   },
```

Comment:
- In this one "We see that Google Play is using a shorthand for millions." other way:
```Python
def convert_string_to_numeric(s):
    if s[-1] == 'M':
        return float(s[:-1]) * 1000000
    else:
        return float(s)
    return np.NaN
test_string = '4.0M'
convert_string_to_numeric(test_string) == 4000000
```
- "What is that badass value? Enter it in the next cell and calculate the proportion of its occurence to the total number of records of google_play." The proportion should be around 15.6% instead of 18.5%. There's something  that is not working properly in your code, I guess the denominator. One way of doing this stuff could be: 
```Python
odd_value = "Varies with device"
google_play[google_play.Size == odd_value].shape[0] / google_play.shape[0]
```
- In this one "Now, use the value_counts() function (documentation) to get a sense on how many apps are in this dataset and their review counts.", just get the number of Apps and their review counts.
```Python
#Try this instead
review_missing_removed.App.value_counts()
```
- In this one "Step 1: Count the number of positive reviews." you can do as simple as the following: 
```Python
sum(np.where(x == "Positive", 1, 0))
```
- Other way of "Make sure to account for the case where the denominator is zero using the np.where() function." could be: 
```Python
google_agg['Positive Ratio'] = np.where(google_agg['Total'] !=0, google_agg['Positive'] / google_agg['Total'], 0)
```